ni_measurementlink_service/_internal/parameter/_message.py

```diff
@@ -0,0 +1,113 @@
+import struct
+from typing import Any, Dict
+
+from google.protobuf.internal import encoder, wire_format
+from google.protobuf.message import Message
+
+from ni_measurementlink_service._internal.parameter._serializer_types import (
+    Decoder,
+    Encoder,
+    Key,
+    NewDefault,
+)
+
+
+def _message_encoder_constructor(field_index: int, is_repeated: bool, is_packed: bool) -> Encoder:
+    """Mimics google.protobuf.internal.MessageEncoder.
+
+    We cannot use MessageEncoder because it is for the pure-Python implementation
+    of messages. Our messages are upb messages, so we need to call 'SerializeToString'
```

Comment:
> Our messages are upb messages

It's not up to us.

Also, "our messages" is confusing. I think you are refering to DoubleXY and not the configuration/output parameters?

```suggestion
    This function was forked in order to call SerializeToString instead of _InternalSerialize.
    
    _InternalSerialize is only defined for the pure-Python protobuf implementation. Our child messages
    (like DoubleXY) are defined in .proto files, so they use whichever protobuf implementation that
    google.protobuf.internal.api_implementation chooses (usually upb).
```